### PR TITLE
Refactor with SOLID principles

### PR DESCRIPTION
**Pull request by bengigo**
**Changes made on this branch:**
- Refactor the main.rb to only start the application.
- Move the pieces of logic to the App class

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -14,6 +14,16 @@ def initialize
...
+  def list_actions
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
```
Comment (Davidosky007) [inline app.rb]: * As stated in the [project requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_refactor.md#:~:text=Make%20sure%20that%20the%20role%20of%20main.rb%20is%20only%20to%20start%20your%20code%20and%20ask%20the%20user%20for%20the%20initial%20choice%20of%20options%20(or%20to%20quit).) `Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).` it is a requirement to follow the SOLID principle

File: app.rb [hunk 2]
```diff
@@ -50,20 +62,23 @@ def create_rental
     date = gets.chomp
     @rentals << Rental.new(@people[person_index], @books[book_index], date)
     puts 'Rental created successfully'
+    run
   end
 
   def list_books
-    return puts 'No books found!' if @books.empty?
```
Comment (Davidosky007) [inline app.rb:56]: * I get these bugs when I try to create a person and a rental 
* When I select list all people the name of the person I created doesn't show up also for creating rentals when I select a book from the list by the number I get invalid input. Kindly fix them 
image
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @bengigo @a-laarabi 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Proper use of Gitflow 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**